sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -332,31 +332,27 @@ def initialize_connection(self):
     def make_metric_list_to_collect(self):
         # Pre-process the list of metrics to collect
         try:
-            # check to see if the database exists before we try any connections to it
-            db_exists, context = self.connection.check_database()
-            if db_exists:
-                if self.instance.get('stored_procedure') is None:
-                    with self.connection.open_managed_default_connection():
-                        with self.connection.get_managed_cursor() as cursor:
-                            self.autodiscover_databases(cursor)
-                        self._make_metric_list_to_collect(self._config.custom_metrics)
-            else:
-                # How much do we care that the DB doesn't exist?
-                ignore = is_affirmative(self.instance.get("ignore_missing_database", False))
-                if ignore is not None and ignore:
-                    # not much : we expect it. leave checks disabled
-                    self.do_check = False
-                    self.log.warning("Database %s does not exist. Disabling checks for this instance.", context)
-                else:
-                    # yes we do. Keep trying
-                    msg = "Database {} does not exist. Please resolve invalid database and restart agent".format(
-                        context
-                    )
-                    raise ConfigurationError(msg)
-
-        except SQLConnectionError as e:
-            self.log.exception("Error connecting to database: %s", e)
-        except ConfigurationError:
+            if self._config.ignore_missing_database:
+                # self.connection.check_database() will try to connect to 'master'.
+                # If this is a DB hosted on Azure the function would throw.
```

Comment:
`DB hosted on Azure` is too vague. Let's be specific that the issue is for Azure SQL Database.